oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/worker/MetricsPersistentWorker.java

```diff
@@ -209,15 +211,11 @@ private void loadFromStorage(List<Metrics> metrics) throws IOException {
             context.clear();
         }
 
-        List<String> notInCacheIds = new ArrayList<>();
-        for (Metrics metric : metrics) {
-            if (!context.containsKey(metric)) {
-                notInCacheIds.add(metric.id());
-            }
-        }
-
-        if (notInCacheIds.size() > 0) {
-            List<Metrics> metricsList = metricsDAO.multiGet(model, notInCacheIds);
+        List<Metrics> notInCacheMetrics = metrics.stream()
+                                                 .filter(m -> !context.containsKey(m))
+                                                 .collect(Collectors.toList());
```

Comment:
```java
final Stream<Metrics> metricsStream = metrics.stream().filter(m -> !context.containsKey(m));
```

I think this stream is better choice for `metricsDAO#multiGet`, WDYT?

Reply:
This could avoid duplicated `for` loop before and inside `metricsDAO#multiGet`

Reply:
Passing the `stream` as an argument is very dangerous. Because `stream` can be consumed only one time.